src/libs/actions/Report.js

```diff
@@ -495,6 +495,85 @@ function fetchChatReports() {
         });
 }
 
+/**
+ * Get a simplified version of an IOU report
+ *
+ * @param {Number} reportID
+ * @param {Object} reportData
+ * @param {Number} reportData.transactionID
+ * @param {Number} reportData.amount
+ * @param {String} reportData.currency
+ * @param {String} reportData.created
+ * @param {String} reportData.comment
+ * @param {Object[]} reportData.transactionList
+ * @param {String} reportData.ownerEmail
+ * @param {String} reportData.managerEmail
+ * @returns {Object}
+ */
+function getSimplifiedIOUReport(reportID, reportData) {
+    const transactions = _.map(reportData.transactionList, transaction => ({
+        transactionID: transaction.transactionID,
+        amount: transaction.amount,
+        currency: transaction.currency,
+        created: transaction.created,
+        comment: transaction.comment,
+    }));
+
+    return {
+        reportID,
+        ownerEmail: reportData.ownerEmail,
+        managerEmail: reportData.managerEmail,
+        currency: reportData.currency,
+        transactions,
+    };
+}
+
+/**
+ * Fetches the updated data for an IOU Report and updates the IOU collection in ONYX
+ *
+ * @param {Number} reportID
+ * @param {Object[]} reportHistory
+ */
+function updateIOUReportData(reportID, reportHistory) {
+    const containsIOUAction = _.any(reportHistory, action => action.actionName === 'IOU');
+
+    // If there aren't any IOU actions, we don't need to fetch any additional data
+    if (!containsIOUAction) {
+        return;
+    }
+
+    const otherParticipants = _.chain(reportHistory)
+        .pluck('actorEmail')
+        .unique()
+        .without(currentUserEmail)
+        .value();
+
+    // If we have more than one participant, this is not an IOU
+    if (otherParticipants.length > 1) {
+        return;
+    }
+
+    // If we have an IOU action, get the IOU reportID
+    let iouReportID = 0;
+    API.GetIOUReport({
+        debtorEmail: otherParticipants[0],
+    }).then((response) => {
+        iouReportID = response.reportID;
+
+        Log.info('[Report] Fetching IOU report data', true, iouReportID);
+        return API.Get({
+            returnValueList: 'reportStuff',
+            reportIDList: iouReportID,
+            shouldLoadOptionalKeys: true,
+            includePinnedReports: true,
+        });
+    }).then((response) => {
+        const iouReportData = response.reports[iouReportID];
+        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportID}`,
```

Comment:
Is there a good reason to put these into their own Onyx collection? I kind of think that they should just be treated like normal reports. If anything, we should maybe add a `type` property to the report collection to be able to tell the different report types apart in the future (I think this will scale better as we get more report types). Note: this will also require writing a migration so that chat reports are stored with the proper `type` and we _maybe_ need to filter the LHN so that it only displays reports with the chat type?

Reply:
So wherever we'd use the REPORT_IOUS collection we could just connect to the REPORT collection instead and filter by type in the callback? This makes sense to me, @Julesssss is there a reason why a new collection would work better here?

Reply:
What's the benefit? I thought this was better because it allowed us to subscribe to just IOUs... are you suggesting we save them with a key like `REPORTS_${type}_${reportID}`?

Reply:
The idea was that keeping a layer of separation between them led to a simpler implementation, as both the method of retrieving and the persisted fields are quite different for Chat/Reports. That was the only reason though.

I'd like to think this through a bit first, to make sure we don't create new problems with this approach.

Reply:
> What's the benefit? 

This same question can be asked of the current implementation. Currently, I don't see that we are subscribing to a list of all IOU reports, we are only accessing a single one. Are there plans to subscribe to the whole list of _just_ IOU reports? If not, then I don't think we need to optimize for that case.

> are you suggesting we save them with a key like REPORTS_${type}_${reportID}?

No, I wasn't suggesting that because this is the current implementation. I'm suggesting that every report (regardless of type) be stored as `REPORT_${reportID}` and then each report has a property for `type`. 

> One concern is that we end up with a mega collection of every report, chat, invoice, IOU... and that we'll make each of these individual cases slower.

I don't see this as a real concern or it's a pre-optimization. Those cases don't exist today and we shouldn't speculate about what would slow them down or not. It's not hard to slap a `_.findWhere(reports, {type:'chat'})` in the places that you need specific report types.

My biggest concern is with this getting out of hand at some point. I am worried that will can lead to a separate action file for each report type that still shares 95% of things in common with each other and a whole bunch of different Onyx collections whose objects are 95% the same. I think that will make the entire system more difficult to learn and maintain over time.

Reply:
Yeah, I don't know what will be best, since we don't know what future uses we will have.
We should do what's easier to change later (which honestly I have no idea what that is).

Reply:
I think if I had to pick an "easier" path, it's probably the one we are currently on because it would mean that no Onyx migration needs to be written in this PR and doesn't require any additional code changes for something that none of us are totally clear on.

Reply:
Since we're currently only fetching [chatReports](https://github.com/Expensify/Expensify.cash/blob/ab38d478ac912b06529bd0e1b1eb5381cbdce8e2/src/libs/actions/Report.js#L479), if we decide to keep everything in reports we'll have to update that logic on the backend to now fetch IOUs and Chat reports, and add necessary type filtering throughout the codebase. Then later on, if we want to migrate to moving everything to a `reportIOU` collection, we'd have to write a migration that removes any IOU reports from Onyx, re-fetches them via GetReportStuff (or some other method), and then inserts the simple structure into Onyx for that collection key. Additionally, we'd have to go through and clean up any filtering logic.

Alternatively, if we keep the `reportIOU` collection and later decide to move everything back into the report collection, we'd need to make the back-end updates and add filtering, as well as add a migration that would remove the reportIOU collection from Onyx, re-fetch all of those reports and add the simplified data to the report collection.

Both of these require some work and a migration, but both options are reasonable/doable IMO. I understand @tgolen's concern about having different collections for different report types that could easily hold the same data, so my main question is if there are any other reasons to use a new collection other than the fear that using the report collection will slow things down (other than the fact that this code is written already).

Reply:
>  so my main question is if there are any other reasons to use a new collection

The only other reason I see is that few fields will be shared for now. There is no need to persist the full IOU report and perhaps option B would require us to do this to align with reports? 

I lean towards the current solution mainly to speed up the IOU project progress, but don't feel too strongly either way. Let's decide on one path and keep moving forward. Please choose from **A** or **B**:

**A:** Keep separate collection for now:
- Quicker and arguably simpler *for now
- A pre-optimization
- Persisted IOUReport data remains lean
- No Onyx migration *for now

**B:** Merge IOU data with the report collection:
- Leads to more shared functionality between Report types
- Better solution in the long run
- Will need to update backend logic to now fetch IOUs and Chat reports
- Might need to persist full IOU Report data (unconfirmed, not a huge problem)
- Requires Onyx migration

Reply:
After bringing this conversation to [slack](https://expensify.slack.com/archives/C011W8BJ9L6/p1616003536027800), the general consensus was that we'll go with option A here.